flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -722,9 +723,11 @@ class ExprLowering {
         const auto &lb = std::get<1>(pair);
         idx.push_back(builder.createIntegerConstant(getLoc(), idxTy, dim - lb));
       }
+      auto insVal = builder.createConvert(getLoc(), eleTy, constant);
```

Comment:
Out of curiosity, was the type mismatch a `fir.logical` vs `i1` or a `fir.real` vs `mlir.float` ?

Reply:
It was building an array of 32-bit logicals with true and false.